src/providers/formattingProvider.utils.ts

```diff
@@ -1,3 +1,57 @@
+import * as vscode from 'vscode';
+
+/**
+ * Build indentation chars
+ * @param depth Depth of indent
+ * @param options VS Code formatting options
+ */
+export function buildIndentationChars(
+    depth: number,
+    options: vscode.FormattingOptions,
```

Comment:
This way, when we are creating mock data, we only have to mock the `tabSize` property instead of all the properties

```suggestion
    options: Pick<vscode.FormattingOptions, 'tabSize'>
```

Reply:
But this function needs two property: `tabSize` and `insertSpaces`. 

Change this:

`buildIndentedLine(..., options) -> buildIndentationChars(..., options)`

to this:

`buildIndentedLine(..., tabSize, insertSpaces) -> buildIndentationChars(..., tabSize, insertSpaces)` ?

Reply:
Not sure how I missed `insertSpaces`, anyway, `Pick` can support any number of properties, so let's just do this:

```suggestion
    options: Pick<vscode.FormattingOptions, 'insertSpaces' | 'tabSize'>,
```